src/renderer/components/shared/CanvasDataBuilder.js

```diff
@@ -498,12 +524,29 @@ const buildNeighbours = async (targetConcept, answers) => {
 
   data = deduplicateConcepts(data);
 
-  const nodes = data.filter(item => item.shouldVisualise).map(item => getNeighbourNode(item.concept, item.graqlVar, item.explanation, item.queryPattern));
+  const nodes = (await Promise.all(data.filter(item => item.shouldVisualise).map(item => getNeighbourNode(item.concept, item.graqlVar, item.explanation, item.queryPattern))))
+    .reduce(collect, []);
   const edges = (await Promise.all(data.filter(item => item.shouldVisualise).map(item => getNeighbourEdges(item.concept, targetConcept)))).reduce(collect, []);
 
   return { nodes, edges };
 };
 
+const getNodesWithUpdatedLabel = async (nodes, type) => {
```

Comment:
errr this being a `get` sounds like it's retrieving something, but this is just updating the label right? 
it might look a tad cleaner if:
1. you had a method on your `facade` that does a `getNodesWithType(type)` instead of just `getAllNodes()`
2. pass the filtered nodes in here
3. name this method `updateNodeLabels(nodes)`, so it's not doing both a filter and an update

what do you think?

Reply:
I agree, that'd be cleaner. however, for number 1, I think it'd be better if the filtering of nodes by type is done directly in the consumer of `updateNodesLabel` (here being the `UPDATE_NODES_LABEL` action in `src/renderer/components/Visualiser/store/actions.js`) immediately before the call to `updateNodesLabel`. mainly because `CDB` doesn't currently have access to `state.visFacade` and so wouldn't be able to get all currently visualised nodes.

so from the outside (in this case, the action handler), it'll look like this:

```
const nodesToUpdate = state.visFacade.getAllNodes().filter(x => x.type === type);
const updatedNodes = await CDB.updateNodesLabel(nodesToUpdate);
state.visFacade.updateNode(updatedNodes);
```

addressed in https://github.com/graknlabs/workbase/pull/284/commits/0c95605a67f7f7a61a26b311d398990f1cd88424

Reply:
cheers :)